tests/src/Unit/SocialApiTest.php

```diff
@@ -0,0 +1,560 @@
+   * test for class SocialApiDataHandler
+   */
+
+  public function testSocialApiDataHandler () {
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/SocialApiDataHandler.php');
+    $this->session = $this->getMock(SessionInterface::class);
+    $collection = $this->getMockBuilder('Drupal\social_api\SocialApiDataHandler')
+                       ->setConstructorArgs(array($this->session))
+                       ->getMockForAbstractClass();
+
+    // checking for correct setSessionPrefix and getSessionPrefix method
+    $collection->setSessionPrefix('1234');
+    // var_dump($collection->getSessionPrefix());
+    $this->assertEquals('1234_', $collection->getSessionPrefix());
+  }
+
+  /**
+   * test for class SocialApiController
+   */
+
+  public function testSocialApiController () {
+
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/Controller/SocialApiController.php');
+
+    $this->namespaces = $this->createMock(Traversable::class);
+    $this->cache_backend = $this->createMock(CacheBackendInterface::class);
+    $this->module_handler = $this->createMock(ModuleHandlerInterface::class);
+
+    // passing parameters for constructor method in NetWorkManager Class
+   $this->networkManager = $this->getMockBuilder('Drupal\social_api\Plugin\NetworkManager')
+                                 ->setConstructorArgs(array($this->namespaces, $this->cache_backend, $this->module_handler))
+                                 ->getMock();
+
+
+   $collection = $this->getMockBuilder('Drupal\social_api\Controller\SocialApiController')
+                                ->setConstructorArgs(array($this->networkManager))
+                                ->getMock();
+
+   $this->assertTrue(
+      method_exists($collection, 'create'),
+      'SocialApiController does not have getPluginId function/method'
+    );
+
+   $this->assertTrue(
+      method_exists($collection, 'integrations'),
+      'SocialApiController does not have getPluginId function/method'
+    );
+    $expected = $collection->integrations($this->type);
+  }
+
+  /**
+   * Test for class SocialApiException
+   */
+
+  public function testSocialApiException () {
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/SocialApiException.php');
+    $collection = new SocialApiException();
+    $this->assertTrue($collection instanceof SocialApiException);
+  }
+
+  /**
+   * Test for class SocialApiImplementerInstaller
+   */
+
+  public function testSocialApiImplementerInstaller () {
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/Utility/SocialApiImplementerInstaller.php');
+
+    $collection = new SocialApiImplementerInstaller();
+    // mock object to our interface
+    $mock = $this->getMock('Drupal\social_api\Utility\SocialApiImplementerInstaller');
```

Comment:
This type of tests is silly. You're not testing anything. Of course the mock will be an instance of that class. You literally said so in the previous line. Anyways, you don't need to add any tests to this file, so you can remove the test.